libs/core/src/components/datepicker/datepicker.test.ts

```diff
@@ -513,5 +514,34 @@ describe('<gds-datepicker>', () => {
         onlyDate(new Date('2024-01-01')),
       )
     })
+
+    it('should not overflow the year when trying to increase max year', async () => {
+      const el = await fixture<GdsDatepicker>(
+        html`<gds-datepicker value="2034-12-10" min="2014-01-01" max="2034-12-31" open></gds-datepicker>`,
+      )
+      await el.updateComplete
+
+      const yearDropdown = el.shadowRoot!.querySelector<GdsDropdown>(
+        `${getScopedTagName('gds-dropdown')}[label="Year"]`,
+      )!
+      
+      let keyPress = 'Shift+Tab'
+      if (isWebKit()) {
+        keyPress = 'Shift+Alt+Tab'
+      }
+      
+      await sendKeys({
+        press: keyPress,
+      })
+    
+      await sendKeys({
+      press: 'Enter',
+      })
```

Comment:
Here, I think it would be easier to just target the next month button directly.

Something like this:
```ts
const nextMonthButton = el.shadowRoot!.querySelector<GdsButton>(
  `${getScopedTagName('gds-button')}[aria-label="Next month"]`,
)!

nextMonthButton.click()
```

That will also make it easier to write a test that checks the previous month action.

Reply:
Fixed